CoreUtilities/src/au/gov/asd/tac/constellation/utilities/rest/RestClient.java

```diff
@@ -95,6 +108,54 @@ public static URL generateUrl(final String url, final Map<String, String> params
         return new URL(url + (query.length() > 0 ? "?" + query : ""));
     }
 
+    /**
+     * Construct a URL string based on supplied URL and any supplied query
+     * parameters.
+     * 
+     * @param url URL to base generated URL on.
+     * @param params Any parameters to add to the URL (ie HTTP GET parameters)
+     * @param paramSpaceToken Allow the token used to replace spaces in param
+     * strings to be adjusted
+     * @return URL combining base URL and parameters.
+     * @throws UnsupportedEncodingException
+     * @throws MalformedURLException
+     */
+    public static URL generateUrl(final String url, final List<Tuple<String, String>> params, final String paramSpaceToken) throws UnsupportedEncodingException, MalformedURLException {
+        // Build the request parameters.
+        final StringBuilder query = new StringBuilder();
+        if (params != null) {
+            for (final Tuple<String, String> param : params) {
+                // Ensure the parameter has a non empty key. Rules for parameter names seem quite relaxed in HTTP, as
+                // such we wil not try and to too much validation
```

Comment:
I assume it's meant to say "do too much" rather than "to too much"

Reply:
done